### DISCUSSION TITLE
WGSL Group and Binding Attributes

### DISCUSSION THREAD

**Discussion by ghadeeras** [gpuweb/gpuweb#2661] [Q&A] [answered]
Hello!

The WGSL specs state that:

> Each resource variable must be declared with both group and binding attributes.

I am just curious, why not then have a single attribute for both? Something like: @resource(group, binding) ... Is there a case where one attribute could be used without the other?

**Comment by Kangz** [comment, marked as answer]
Both attributes have to be used, and the idea of merging them has been raised repeatedly. What prevent it from going forward is that when reading `@resource(2, 1)`, how do you know which is the group and which is the binding?

**Reply by ghadeeras** [reply]
I was reluctant to ask because I thought this must've been raised repeatedly :-) I was just curious if there was some usage I was not aware of that mandated the separation of the two attributes, but it seems it is mainly for readability.  Thanks @Kangz !
